test/utils/utils.cpp

```diff
@@ -3,21 +3,20 @@
 // Licensed under the Apache License, Version 2.0.
 
 #include <test/utils/utils.hpp>
-
-#include <iomanip>
-#include <iostream>
 #include <regex>
+#include <stdexcept>
 
 bytes from_hex(std::string_view hex)
 {
     if (hex.length() % 2 == 1)
         throw std::length_error{"the length of the input is odd"};
 
     bytes bs;
+    bs.reserve(hex.length() / 2);
     int b = 0;
     for (size_t i = 0; i < hex.size(); ++i)
     {
-        auto h = hex[i];
+        const auto h = hex[i];
         int v;
         if (h >= '0' && h <= '9')
```

Comment:
somewhere deeper down in the else-branch, there is a throw statement that isn't documented in the header file (as the rest is). Would it make sense to state that?

Reply:
Functions are not documented.